### PR TITLE
11085: Pre PinotConfig commons-configuartions2 upgrade 

### PR DESCRIPTION
**Pull request by abhioncbr**
- As per the [issue](https://github.com/apache/pinot/issues/11085), This is the pre-steps for upgrading the `PinotConfiguration` to commons-configuration2. 
- In continuation of the last [PR](https://github.com/apache/pinot/pull/11792) 

cc: @Jackie-Jiang / @xiangfu0 / @walterddr

### CODE REVIEW

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TlsIntegrationTest.java [hunk 1]
```diff
@@ -211,7 +211,7 @@ protected PinotConfiguration getDefaultServerConfiguration() {
         CertBasedTlsChannelAccessControlFactory.class.getName());
     prop.put("pinot.server.adminapi.access.protocols", "internal");
     prop.put("pinot.server.adminapi.access.protocols.internal.protocol", "https");
-    prop.put("pinot.server.adminapi.access.protocols.internal.port", "7443");
+    prop.put("pinot.server.adminapi.access.protocols.internal.port", "9443");
```
Comment (abhioncbr) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TlsIntegrationTest.java:214]: I always get the issue in binding to this port on my local Mac machine with Intellij, Is it okay to change the port to 9443?

Comment (xiangfu0) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TlsIntegrationTest.java:214]: I plan to randomize ports for tests: https://github.com/apache/pinot/pull/11861

Comment (abhioncbr) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TlsIntegrationTest.java]: Reversed my change. Thanks

File: pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java [hunk 2]
```diff
@@ -101,10 +101,16 @@ private static Iterable<String> getIterable(Iterator<String> keys) {
    * @param configuration to iterate on keys
    * @return a stream of keys
    */
+  @Deprecated
   public static Stream<String> getKeysStream(Configuration configuration) {
     return StreamSupport.stream(getIterable(configuration.getKeys()).spliterator(), false);
   }
 
+  public static Stream<String> getKeysStream(org.apache.commons.configuration2.Configuration configuration) {
```
Comment (xiangfu0) [inline pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java:109]: Put `org.apache.commons.configuration2.Configuration` to import section and make the deprecated `org.apache.commons.configuration.Configuration` to full name.

**Review by xiangfu0 — APPROVED** [review]

**Pull request merged**